### PR TITLE
b/150729554: enable ignoredQueryParameters and ignoredUnknownQueryPar…

### PR DESCRIPTION
**Pull request by TAOXUY**
…ameters

### CODE REVIEW

File: src/go/configgenerator/listener_generator.go [hunk 1]
```diff
@@ -639,27 +639,16 @@ func makeTranscoderFilter(serviceInfo *sc.ServiceInfo) *hcmpb.HttpFilter {
 				DescriptorSet: &transcoderpb.GrpcJsonTranscoder_ProtoDescriptorBin{
 					ProtoDescriptorBin: configContent,
 				},
-				AutoMapping:       true,
-				ConvertGrpcStatus: true,
+				AutoMapping:                  true,
+				ConvertGrpcStatus:            true,
+				IgnoredQueryParameters:       transcodingIgnoreQueryParameters(serviceInfo),
+				IgnoreUnknownQueryParameters: serviceInfo.Options.TranscodingIgnoreUnknownQueryParameters,
```
Comment (qiwzhang) [inline src/go/configgenerator/listener_generator.go]: If IgnoreUnknownQueryParameters is true,  we don't need to set another one.  It doesn't make sense to set both.

Comment (TAOXUY) [reply src/go/configgenerator/listener_generator.go]: Good point, will add exception in start_esp.py

File: src/go/configgenerator/listener_generator.go [hunk 2]
```diff
@@ -799,3 +788,38 @@ func makeRouterFilter(opts options.ConfigGeneratorOptions) *hcmpb.HttpFilter {
 	}
 	return routerFilter
 }
+
+func transcodingIgnoreQueryParameters(serviceInfo *sc.ServiceInfo) []string {
```
Comment (JLXIA) [inline src/go/configgenerator/listener_generator.go]: move this logic into serviceinfo.go, and considering to consolidate TranscoderIgnoredJwtQueryParams, TranscoderIgnoredApiKeyQueryParams, transcodingIgnoreQueryParameters.

File: 1 [hunk 3]
```diff
@@ -0,0 +1,2122 @@
+..usage: python3 -m unittest [-h] [-s SERVICE] [-v VERSION]
```
Comment (nareddyt) [inline 1]: Remove file

**Comment by TAOXUY** [comment]
PTAL

**Pull request merged**